src/silx/app/view/Viewer.py

```diff
@@ -305,7 +305,7 @@ def __synchronizeH5pyObject(self, h5):
         # FIXME: The architecture have to be reworked to support this feature.
         # model.synchronizeH5pyObject(h5)
 
-        filename = h5.filename
+        filename = f"{h5.file.filename}::{h5.name}"
```

Comment:
This need some reworks: It would be better to keep the initial url used to reload (it would also work for #3677)